File: pset6/exercises/cash/cash.py
from cs50 import get_float

def printCoins(money):

    cents, total = [25, 10, 5, 1], 0

    for i in range(4):

        coins = int(money / cents[i])
        money = money - (cents[i] * coins)
        total = total + coins

        if money == 0: break

    print(total)


while True:

    cash = get_float("Change owed: ")
    if cash > 0: break

cash = int(cash * 100)
printCoins(cash)

File: pset1/exercises/cash/cash.c
#include <stdio.h>
#include <math.h>
#include <cs50.h>

float get_value(void);
int calculate_coins(int cash, int value);

int main(void)
{
    float dollars = get_value();
    int cents = round(dollars * 100);

    int values[] = {25, 10, 5, 1};
    int coins = 0;

    for (int i = 0; i < 4; i++)
    {
        int value = values[i];

        int quantity = calculate_coins(cents, value);
        cents = cents - quantity * value;

        coins += quantity;
    }

    printf("%i\n", coins);
}

float get_value(void)
{
    float value;
    do
    {
        value = get_float("Troco: ");
    }
    while (value <= 0);

    return value;
}

int calculate_coins(int cash, int value)
{
    int coins = cash / value;
    return coins;
}